### PR TITLE
Sc/improve optimismportal test cov

### PR DESCRIPTION
**Pull request by AmadiMichael**
improve optimismPortal(2) test coverage

### CODE REVIEW

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol [hunk 1]
```diff
@@ -968,7 +968,7 @@ contract OptimismPortal_FinalizeWithdrawal_Test is CommonTest {
     }
 
     /// @dev Tests that `finalizeWithdrawalTransaction` succeeds when _tx.data is empty.
-    function test_finalizeWithdrawalTransaction_noTxData_succeeds() external {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/OptimismPortal.t.sol]: We probably want both tests, no?

File: packages/contracts-bedrock/test/L1/OptimismPortal2.t.sol [hunk 2]
```diff
@@ -439,6 +439,12 @@ contract OptimismPortal2_FinalizeWithdrawal_Test is CommonTest {
     constructor() {
         super.setUp();
 
+        // zero out contracts that should not be used
+        assembly {
```
Comment (smartcontracts) [inline packages/contracts-bedrock/test/L1/OptimismPortal2.t.sol]: Honestly I think we can leave this out for now, if we're going to do stuff like this I would do it in a different PR after opening an issue to discuss the reasoning for it

File: packages/contracts-bedrock/test/L1/OptimismPortal.t.sol (hunk 1 above)
Comment (AmadiMichael) [reply packages/contracts-bedrock/test/L1/OptimismPortal.t.sol]: we do actually

**Review by smartcontracts — APPROVED** [review]
